packages/orchestrator/lib/clients/processor.ts

```diff
@@ -121,26 +122,29 @@ export class OrchestratorProcessor {
                         // task was aborted. No need to set it as failed
                         return;
                     }
+
                     const setFailed = await this.orchestratorClient.failed({ taskId: task.id, error: res.error });
                     if (setFailed.isErr()) {
-                        logger.error(`failed to set task ${task.id} as failed: ${stringifyError(setFailed.error)}`);
+                        throwIfPayloadTooBig(setFailed.error);
```

Comment:
not a huge fan of now having 2 error handling paths. can we instead check the error and call `orchestratorClient.failed()` again with a `payload_too_big` error.

Reply:
that's basically what it does, throwing will call `failed()` with the error name, and I don't have to copy paste 2 more times the call + isErr()
let me know if you still fell like it's worth the added cost

Reply:
I see your point. I wonder if this is not the right level to handle this error, exposing too much of the internals of the orchestrator. What if `.failed()` `.succeed()` and all functions passing a payload were dealing with the payload to big error internally, calling `failed()` themselves.

Reply:
Definitely a weird place to handle it, but also I can't really see better place because the only function around processTask is processingLoop and it's generic too
nb: it's the only place where it matters because we still want to mark the task as failed no matter the input size.

Reply:
the other place would be in the [orchestrator](https://github.com/NangoHQ/nango/blob/525e34011f3612acca6813023e40d92c816bed05/packages/orchestrator/lib/routes/v1/tasks/putTaskId.ts#L54), no exposing the pg payload_too_big error. I understand if you don't want to take care of this in this PR, if not add a TODO and a ticket and assign it to me

Reply:
the thing is this error pops before entering the actual endpoint (because it's a middleware), so it's tricky

Reply:
I'll create a linear issue

Reply:
in the orchestrator client then: https://github.com/NangoHQ/nango/blob/525e34011f3612acca6813023e40d92c816bed05/packages/orchestrator/lib/clients/client.ts#L383

Still a level higher and "users" of the client won't have to deal with this internal error